discovery-provider/src/tasks/update_track_is_available.py

```diff
@@ -0,0 +1,250 @@
+import logging
+from datetime import datetime
+from typing import Any, List, Tuple, TypedDict, Union
+
+import requests
+from src.models import Track, URSMContentNode, User
+from src.tasks.celery_app import celery
+
+logger = logging.getLogger(__name__)
+
+UPDATE_TRACK_IS_AVAILABLE_LOCK = "update_track_is_available_lock"
+ALL_UNAVAILABLE_TRACKS_REDIS_KEY = "update_track_is_available:unavailable_tracks_all"
+BATCH_SIZE = 1000
+DEFAULT_LOCK_TIMEOUT_SECONDS = 86400  # 24 hour -- the max duration of 1 worker
+REQUESTS_TIMEOUT_SECONDS = 300  # 5 minutes
+
+
+class ContentNodeInfo(TypedDict):
+    endpoint: str
+    spID: int
+
+
+def _get_redis_set_members_as_list(redis: Any, key: str) -> List[int]:
+    values = redis.smembers(key)
+    return [int(value.decode()) for value in values]
+
+
+def fetch_unavailable_track_ids_in_network(session: Any, redis: Any) -> None:
+    content_nodes = query_registered_content_node_info(session)
+
+    # Clear redis for existing data
+    redis.delete(ALL_UNAVAILABLE_TRACKS_REDIS_KEY)
```

Comment:
i think using redis adds unnecessary complexity. discussed offline + don't want to block on this.